oci/oci.go

```diff
@@ -128,12 +128,25 @@ func (r *Runtime) CreateContainer(c *Container) error {
 	childPipe.Close()
 
 	// Wait to get container pid from conmon
-	// TODO(mrunalp): Add a timeout here
-	var si *syncInfo
-	if err := json.NewDecoder(parentPipe).Decode(&si); err != nil {
-		return fmt.Errorf("reading pid from init pipe: %v", err)
+	s := make(chan *syncInfo, 1)
+	e := make(chan error, 1)
```

Comment:
can we use a struct here? instead of having two channels? Something like:
```go
	type syncStruct struct {
		si  *syncInfo
		err error
	}
	ch := make(chan syncStruct) // no need for buffered chan I guess
	go func() {
		var si *syncInfo
		if err := json.NewDecoder(parentPipe).Decode(&si); err != nil {
			ch <- syncStruct{err: err}
			return
		}
		ch <- syncStruct{si: si}
	}()

	select {
	case ss := <-ch:
		if ss.err != nil {
			return err
		}
		logrus.Infof("Received container pid: %q", ss.si.Pid)
	case <-time.After(3 * time.Second): // move this to a constant
		// whatever...
	}
```